lint cleanup

pkienzle opened the pull request:
Additional code and doc cleanup to be applied after PR #408 (issue #407). 

I reformatted the equations for the following models so they align and fit on the page:

* hollow rectangular prism
* spherical sld
* be polyelectrolyte
* mass surface fractal

I introduced a variable substitution into hollow rectangular prism to make the calculation more obvious.  I don't believe we have the requirement `A < B < C` in the math or the code, but I left it in the docs.

smk78 commented:
I have built this branch locally and have skimmed through every single one of the html-ised model docs looking for formatting errors, etc, making some small corrections in the process (see commits above). On the whole, all looks fine. Except:

- hollow_rectangular_prism: This now says

![image](https://user-images.githubusercontent.com/10938679/90024016-53678400-dcac-11ea-8689-ba13b6ad3648.png)

  whereas before it said

![image](https://user-images.githubusercontent.com/10938679/90024190-7e51d800-dcac-11ea-8768-9e0f46c2b3d4.png)

  Q1. Is the assignment of the primed variables correct? ie, Should it be A' = A + 2delta, etc?
  Q2. What happened to the factor of 8 multiplying the second line of the big equation?

pkienzle commented:
A1: `A` is the outer dimension and `A' = A - 2Δ` is the inner dimension, so `A = A' + 2Δ`.

A2: `8(A/2 - Δ)(Β/2-Δ)(C/2-Δ) = [2(Α/2-Δ)][2(B/2-Δ)][2(C/2-Δ)] = [2(A - 2Δ)/2][2(B-2Δ)/2][2(C-2Δ)/2] = A'B'C'`

pkienzle commented:
Maybe: `where $A'$, $B'$, $C'$ are the inner dimensions and $A = A' + 2Δ$, $B = B' + 2Δ$, $C =  C' + 2Δ$ are the outer dimensions, and the volume of the parallelepiped is $V = ABC - A'B'C'$.`

This also removes the spurious requirement that `A ≤ B ≤ C`.

pkienzle commented:
I've fixed the errors so that docs build and updated the text in hollow rectangular prism to make inner and outer dimension defs clearer.  This PR is large but largely trivial. It should be reviewed and merged before more substantial changes are done on sasmodels.

smk78 commented:
Just re-pulled and re-built this branch. Changes to model files look fine.

rozyczko reviewed (approved):
9.67/10 is respectable. My pylint runs on sasmodels show some ~230 issues (win/linux) when run as 
`extra\run_pylint.py` but this here is a great start.
I need to do similar analysis/fixup for the qtgui directory, maybe with `pyflakes` or better with `flake8`.

pkienzle commented on extra/pylint.rc:
```diff
@@ -59,23 +59,29 @@ confidence=
+    unnecessary-pass,
+    using-constant-test,
     #no-self-use,
+    useless-object-inheritance,
```
No, we are still building wx gui against py27.

pkienzle commented on sasmodels/exception.py:
```diff
@@ -47,5 +47,5 @@ def annotate_exception(msg, exc=None):
             arg0 = " ".join((args[0], msg))
             exc.args = tuple([arg0] + list(args[1:]))
-        except:
+        except Exception:
```
I would rather not have switches in the code. Lint is too make code cleaner not uglier. Also, the switches are not consistent across linters, so putting them in the code makes it harder to switch to flake8 for example.

But mostly I'm not convinced that `Exception` is overly broad.
